source/agora/node/Validator.d

```diff
@@ -287,7 +287,8 @@ public class Validator : FullNode, API
         if (auto err = this.ledger.validateBlock(block))
         {
             log.error("Block failed to validate: {}", err);
-            return false;
+            // Maybe the block was already added to the ledger
+            return this.ledger.getBlockHeight() >= block.header.height;
```

Comment:
In this case, shouldn't we add our signature or we're guaranteed that it's already done ?

Reply:
If the block was already added to the ledger by this node then it should be signed by this node already.